### PR TITLE
refactor bootstrap image, bump images

### PR DESCRIPTION
**Pull request by BenTheElder**
- use `COPY` instead of `ADD`... `ADD` is more magical
- copy `create_bazel_cache_rcs.sh` into the image
- split the entrypoint into a runner for dind + bazel caching, and the actual bootstrap entrypoint
  - xref: https://github.com/kubernetes/test-infra/issues/8780, this will let podutils jobs work like: `/usr/local/bin/runner.sh my-test-command` with env `DOCKER_IN_DOCKER_ENABLED`. still sorta gross, but it should help us move forward for now. I may refactor further at some point.
- make the new runner use the baked in `create_bazel_cache_rcs.sh` when we don't have a test-infra checkout (so bazel caching should also work in podutils jobs now)

### CODE REVIEW

**Comment by BenTheElder** [comment]
/area images
/assign @krzyzacy 
I've verified that this builds but I haven't tested it at all yet
/hold

**Comment by krzyzacy** [comment]
I guess push and attach a build log from some job that uses the latest image?

**Comment by BenTheElder** [comment]
SGTM, will do that soonish

**Comment by BenTheElder** [comment]
going to rebase onto https://github.com/kubernetes/test-infra/pull/8883 and just make this a bump PR as well

File: images/bootstrap/Dockerfile [hunk 1]
```diff
@@ -121,7 +121,7 @@ ADD ["./barnacle/barnacle", "/usr/local/bin"]
 # note the runner is also responsible for making docker in docker function if
 # env DOCKER_IN_DOCKER_ENABLED is set and similarly responsible for generating
 # .bazelrc files if bazel remote caching is enabled 
-ADD ["runner", \
-    "/workspace/"]
+COPY ["entrypoint.sh", "runner.sh", "create_bazel_cache_rcs.sh", \
+        "/usr/local/bin/"]
 
-ENTRYPOINT ["/workspace/runner"]
+ENTRYPOINT ["/user/bin/entrypoint.sh"]
```
Comment (BenTheElder) [inline images/bootstrap/Dockerfile]: grrr this should be `/usr/bin/entrypoint.sh`

File: images/bootstrap/runner.sh [hunk 2]
```diff
@@ -13,20 +13,22 @@
...
     echo "Bazel remote cache is enabled, generating .bazelrcs ..."
-    ./test-infra/images/bootstrap/create_bazel_cache_rcs.sh
+    # if we have a test-infra checkout (because bootstrap), use that since
+    # it is newer and this is probably k/k so we can push fixes faster this way
+    # otherwise run the one baked into the image
+    # TODO(bentheelder): someday only support the pod-utils
+    if [[ -d "./test-infra" ]]; then
+        ./test-infra/images/bootstrap/create_bazel_cache_rcs.sh
+    else
+        /usr/local/bin/create_bazel_cache_rcs.sh | bash
```
Comment (dims) [inline images/bootstrap/runner.sh]: we have to pipe the output into bash? (we don't do this on line 28)

File: images/bootstrap/entrypoint.sh [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+
+# get test-infra for latest bootstrap etc
+git clone https://github.com/kubernetes/test-infra
+
+# actually start bootstrap and the job, under the runner (which handles dind etc.)
+/usr/local/bin/runner.sh \
+    ./test-infra/jenkins/bootstrap.py \
+        --job="${JOB_NAME}" \
+        --service-account="${GOOGLE_APPLICATION_CREDENTIALS}" \
+        --upload='gs://kubernetes-jenkins/logs' \
```
Comment (dims) [inline images/bootstrap/entrypoint.sh]: should this be configurable? (i remember some jobs have pr-logs or something)

File: images/bootstrap/runner.sh (hunk 2 above)
Comment (BenTheElder) [reply images/bootstrap/runner.sh]: oops, yeah, I initially thought about curl | bash the latest from test-infra but felt terrible about that and did this instead. forgot to drop the pipe to bash 🤦‍♂️ 

thanks!

File: images/bootstrap/entrypoint.sh (hunk 3 above)
Comment (BenTheElder) [reply images/bootstrap/entrypoint.sh]: you can specify it again and overwrite it IIRC. I didn't want to touch these flags as this is how they were in the image now. ideally we will phase out using ENTRYPOINT for this and just have some jobs use the runner script

File: images/bootstrap/runner.sh (hunk 2 above)
Comment (BenTheElder) [reply images/bootstrap/runner.sh]: fixed

File: images/bootstrap/Dockerfile (hunk 1 above)
Comment (BenTheElder) [reply images/bootstrap/Dockerfile]: fixed

**Comment by BenTheElder** [comment]
rebased, fixed the paths, re-bumped

**Comment by BenTheElder** [comment]
/hold cancel
:eyes:
FYI @cjwagner

**Comment by BenTheElder** [comment]
FYI @neolit123 this is going out now. 🤞

**Comment by BenTheElder** [comment]
uh `updateconfig` did not run? @krzyzacy @cjwagner

**Comment by krzyzacy** [comment]
hummmmmm

**Comment by krzyzacy** [comment]
missed a hook ?! I cannot find any related logs

I'll work on #8892 now
/shrug

**Pull request merged**